src/main/java/us/kbase/userprofile/MongoController.java

```diff
@@ -195,89 +182,96 @@ public UserProfile getProfile(String username) {
+					if(up.getProfile().asJsonNode().isObject()) {
+						replacement.put("profile", Document.parse(up.getProfile().asJsonNode().toString()));
+					} else {
 						throw new RuntimeException("Profile must be an object if defined.");
 					}
 				} else {
 					replacement.put("profile", null);
 				}
-				System.out.println(replacement);
-				profiles.update(
-						new BasicDBObject("user.username",up.getUser().getUsername()),
-						new BasicDBObject("$set",replacement));
+				profiles.updateOne(
+						new Document("user.username", up.getUser().getUsername()),
+						new Document("$set", replacement));
 			} else {
-				DBObject user = new BasicDBObject("username",up.getUser().getUsername())
+				Document user = new Document("username", up.getUser().getUsername())
 									.append("realname", up.getUser().getRealname())
 									.append("thumbnail", up.getUser().getThumbnail());
-				
-				DBObject profile = new BasicDBObject("user",user);
-				if(up.getProfile()!=null) {
-					if(up.getProfile().asJsonNode().isObject())
-						profile.put("profile", JSON.parse(up.getProfile().asJsonNode().toString()));
-					else {
+
+				Document profile = new Document("user", user);
+				if (up.getProfile() != null) {
+					if (up.getProfile().asJsonNode().isObject()) {
+						profile.put("profile", Document.parse(up.getProfile().asJsonNode().toString()));
+					} else {
 						throw new RuntimeException("Profile must be an object if defined.");
 					}
 				} else {
 					profile.put("profile", null);
 				}
-				profiles.insert(profile);
+				profiles.insertOne(profile);
 			}
 		}
 	
 	
 	// assume UserProfile is validated
 	public void updateProfile(UserProfile up) {
 		if(exists(up.getUser().getUsername())) {
-			DBObject user = new BasicDBObject("username",up.getUser().getUsername());
-			if(up.getUser().getRealname()!=null)
+			Document user = new Document("username", up.getUser().getUsername());
+
+			if(up.getUser().getRealname() != null) {
 				user.put("realname", up.getUser().getRealname());
-			if(up.getUser().getThumbnail()!=null)
+			}
+			if(up.getUser().getThumbnail() != null) {
 				user.put("thumbnail", up.getUser().getThumbnail());
+			}
 			
-			DBObject update = new BasicDBObject("user",user);
-			if(up.getProfile()!=null) {
+			Document update = new Document("user", user);
+			if(up.getProfile() != null) {
 				JsonNode profileNode = up.getProfile().asJsonNode();
-				System.out.println(profileNode);
-				if( profileNode.isObject() ) {
-					Iterator<Entry<String, JsonNode>> fields = profileNode.fields();
+
+				if(profileNode.isObject()) {
+					Iterator<Map.Entry<String, JsonNode>> fields = profileNode.fields();
 					while(fields.hasNext()) {
-						Entry<String,JsonNode> e= fields.next();
-						update.put("profile."+e.getKey(), JSON.parse(e.getValue().toString()));
+						Map.Entry<String, JsonNode> e = fields.next();
+						Object val = UObject.transformJacksonToObject(e.getValue(), Object.class);
```

Comment:
```suggestion
						final Map.Entry<String, JsonNode> e = fields.next();
						final Object val = UObject.transformJacksonToObject(e.getValue(), Object.class);
```

Will allow compiler to inline

Reply:
👍